test/e2e/autotls/auto_tls_test.go

```diff
@@ -168,3 +207,24 @@ func waitForCertificateReady(t *testing.T, clients *test.Clients, certName strin
 		t.Fatalf("Certificate %s is not ready: %v", certName, err)
 	}
 }
+
+func waitForNamespaceCertReady(t *testing.T, clients *test.Clients) string {
+	var certName string
```

Comment:
t.Helper(), I guess. But I'd rather this returned an error that you can fail in the test.

Reply:
Done. changed to return an error.

Reply:
Comments were addressed. @vagababov Could you take another look? Thanks :)